sklearn/ensemble/gradient_boosting.py

```diff
@@ -1146,7 +1146,8 @@ def feature_importances_(self):
 
     def _validate_y(self, y):
         self.n_classes_ = 1
-
+        if y.dtype is np.dtype(object):
```

Comment:
@jnothman this is how the code is factored. The regressor implementation is in Base, and overwritten in the classifier.